code/modules/clothing/glasses/hud.dm

```diff
@@ -203,4 +203,150 @@ obj/item/clothing/glasses/hud/security/supergars
+	select_mode()
+	usr << "You switch the HUD to [modes[currentmode]]."
+	add_fingerprint(user)
+	..()
+
+/obj/item/clothing/glasses/hud/atmos/equipped(mob/M, slot)
+	if (slot != slot_glasses)
+		for (var/turf/I in src.images)
+			janitor(I)
+	..()
+
+/obj/item/clothing/glasses/hud/atmos/proc/janitor(var/turf/T)
+//People are gonna ask why this is necessary, and its because otherwise the lag clearing all these at once generates is stupendous. This eases them out much nicer.
+	set background = 1
+	var/image/trash = src.images[T]
+	src.images -= T
+	spawn(0)
+		qdel(trash)
+
+/obj/item/clothing/glasses/hud/atmos/process_hud(var/mob/M)
+	set background = 1
+
+	//Make checks to ensure any of this is necessary
+	if(!M)	return
+	if(!M.client)	return
+	if(modes[currentmode] == "Off") return
+
+	//Clear any no longer in view
+	for (var/turf/I in images)
+		if(!(I in view(M.client)))
+			janitor(I)
+
+
+	//Initialize all the vars we'll need
+	var/client/C = M.client
+	var/Red = 0
+	var/Blue = 0
+	var/Green = 0
+	var/datum/gas_mixture/air = null
+	var/pressure = null
+	var/image/holder = null
+
+	//Check every viewable turf to see if it needs an overlay
+	for (var/turf/T in view(C))
+		Red = 0
+		Blue = 0
+		Green = 0
+		air = T.return_air()
+		pressure = air.return_pressure() //saving a bit of work
+		if (istype(T, /turf/unsimulated/) || istype(T, /turf/space/) || T.density) //Tiles exempt from this display
+			if (T in images) //Any tiles that changed from a simulated one to one of the exempted tiles
+				janitor(T)
+			continue
+
+		if (T in images) //If we already have an image for the turf, lets just update the color
+			holder = images[T]
+			if(modes[currentmode] == "Temperature")
+				if (((air.return_temperature() - basetemp) <= 10) && ((air.return_temperature() - basetemp) >= -10)) continue
+				else if (air.return_temperature() < basetemp)
+					Blue = (255/basetemp) * (basetemp - air.return_temperature())
+				else if (air.return_temperature() > basetemp)
+					Red = (255/basetemp) * ((air.return_temperature() - basetemp)/2)
+
+			else if(modes[currentmode] == "Pressure")
+				if (((pressure - basepressure) <= 20) && ((pressure - basepressure) >= -20)) continue
+				else if (pressure < basepressure)
+					Green = (255/basepressure) * (basepressure - pressure)
+				else if (pressure > basepressure)
+					Red = (255/basepressure) * ((pressure - basepressure))
+					Green = Red
+
+			//Clamp to ensure all vars are in the proper range
+			Clamp(Red, 0, 255)
+			Clamp(Green, 0, 255)
+			Clamp(Blue, 0, 255)
+
+			if (!(Red || Green || Blue)) //Delete if its no longer needed
+				qdel(holder)
+			else
+				holder.color = rgb(Red,Green,Blue,intensity) //Set the new color
+			continue
+
+		else //Make the new image
+
+			holder = new/image()
+			holder.icon = 'icons/effects/effects.dmi'
+			holder.icon_state = "overlay"
+			holder.loc = T
+			holder.layer = T.layer
+
+			if(modes[currentmode] == "Temperature") //MATH!

```

Comment:
As far as I can see, the next few lines is an exact copy of the code above. It'd be better if you put this in a function.